### ISSUE TITLE
Site Editor: consistent name for the Edit buttons

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#47780]
### Description

Follow up to https://github.com/WordPress/gutenberg/pull/47343

Actually, in the Site Editor there are _two_ buttons to switch the editor from 'Browse mode' to 'Edit mode':

- The button at the top of the navigation. Its accessible name is 'Edit' (the button text).
- The preview iframe, which has a `role="button"` attribute and an aria-label 'Editor canvas'.

These buttons do the same thing and should have the same name. The name should clearly communicate the action.
'Editor canvas' isn't ideal. For consistency and clarity, it should be changed to 'Edit'.

### Step-by-step reproduction instructions

- Go to the Site Editor.
- Inspect the source with your browser dev tools.
- Hint: use Chrome dev tools and switch to the 'Accessibility' tab.
- Observe the accessible name of the first button is 'Edit'.
- Observe the accessible name of the iframe with `role="button"` is 'Editor canvas'.

### Screenshots, screen recording, code snippet

The two buttons as announced by VoiceOver:

<img width="1792" alt="Edit button" src="https://user-images.githubusercontent.com/1682452/216975889-d785a494-bebe-4e61-a272-08d7abe86dc0.png">

<img width="1792" alt="Editor canvas button" src="https://user-images.githubusercontent.com/1682452/216975902-0b3678ed-f6e6-4936-a397-0d76a543b8d5.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
You also mentioned that the button text changes depending on screen size?

**Comment by abhi3315** [comment]
> You also mentioned that the button text changes depending on screen size?

As mentioned in https://github.com/WordPress/gutenberg/pull/47343#issuecomment-1401518736 by @afercia 

> I'm not sure what's the reasoning for this. 'Edit' and 'View' are two pretty different logical concepts. Either the button action should always be 'Edit' or always be 'View Editor'.

Actually, this `View Editor` button is actually showing the editor only. I think the labelling `View Editor` is serving its purpose. After clicking `View Editor`, Editor is shown.

![GIF Recording 2023-02-07 at 12 33 30 PM](https://user-images.githubusercontent.com/43412958/217172015-2ed953b5-91db-42bd-be6d-fd2ab31214b9.gif)

**Comment by afercia** [comment]
Ah, so on mobile it's a two-steps flow. I missed that.
1. First, click on a template in the navigation. Visually, nothing appears to happen.
2. Then, click 'View editor'. The template preview is shown.
3. Then, click 'Edit'. The template can now be edited.

To be fair, I think that's a bit confusing. When in the mobile view, I'm not sure what's the purpose of showing the preview in the first place. On desktop, the preview is visible while I navigate through the templates. Instead, on mobile the preview is initially hidden. As a user, when I click 'View editor' I would expect to be in the Editor in edit mode. Why do I need an intermediate step for the preview?

Couple more notes:

1
The WP logo 'button' does different things depending on the UI state. Visually, it's always the same 'button' but its functionality changes. That's arguably a good UI. While this happens also on desktop, it's even more confusing on mobile.

2
When clicking 'Manage all templates' or 'Manage all template parts':

On desktop, the 'Edit' button disappears. That makes totally sense, as there's nothing to 'Edit' in the templates list:

<img width="1054" alt="Screenshot 2023-02-08 at 09 35 28" src="https://user-images.githubusercontent.com/1682452/217480245-4930ab46-ca21-43a0-b6c2-0ae95e9d7095.png">

Instead, on mobile the 'View editor' button is still there:

<img width="850" alt="Screenshot 2023-02-08 at 09 33 12" src="https://user-images.githubusercontent.com/1682452/217480450-042b4b54-bc68-421c-bc5f-8abe04a4b536.png">

That's arguably correct as clicking on 'View editor' doesn't bring me to the Editor in edit mode. Instead, it brings me to the templates list:

<img width="816" alt="Screenshot 2023-02-08 at 09 32 53" src="https://user-images.githubusercontent.com/1682452/217480975-c0989c08-02ad-4375-97e3-c7083c2c571d.png">

In the templates list I would expect to see the 'Add New' button at the top right. Instead, I see the 'Edit' button. Clicking this button doesn't edit anything. Instead, it makes the top bar change and finally shoe the 'Add New' button:

<img width="820" alt="Screenshot 2023-02-08 at 09 34 03" src="https://user-images.githubusercontent.com/1682452/217481815-e8c52c09-e6eb-46aa-894b-c43d996d7abe.png">

Overall, it appears there's still a lot of room for improvement in this flow.

**Comment by abhi3315** [comment]
@afercia I do agree. We have to think of a clear flow in the mobile screens and break down it into smaller tasks

**Comment by benridane** [comment]
* Gutenberg Version: 17.8.0-rc.1
I checked with the latest version of Gutenberg and it seems that the button names are obsolete.
The button names have been replaced with ICON and the problem seems to be resolved.
![IMG_0368](https://github.com/WordPress/gutenberg/assets/15902112/2c31ae29-9e9d-415f-aab0-490d5b21f09e)

**Comment by afercia** [comment]
Thank you @benridane yes that's correct.

To recap the current state of this issue:

Many things have changed in the meantime:
- The buttons to switch the editor from 'view' mode to 'edit' mode are now icon buttons.
- Icon buttons are inherently less accessible than buttons with visible text.
- These buttons are labeled 'Edit' and their accessible name is visually exposed via a tooltip:

![Screenshot 2024-02-23 at 09 16 18](https://github.com/WordPress/gutenberg/assets/1682452/3240451f-1ac5-4f0f-b22d-6c2d51a32378)

- The clickable preview is actually the `<iframe>` element of the editor. Its native `role` is overridden via an ARIA `role` attribute so that it is perceived as a button labeled 'Editor canvas'. Overriding a native role this way is a little hacky and arguable. It is labeled 'Editor canvas', which isn't consistent with other buttons that have the same functionality.

https://github.com/WordPress/gutenberg/blob/599f3365232316242a2f92d8fdb5a05e0b0accaa/packages/edit-site/src/components/block-editor/editor-canvas.js#L55-L58
